File: webapp/app/common/enums/PipelineRunStage.ts
export enum PipelineRunStage {
  RUN_TRIGGERED = "RUN_TRIGGERED",
  RECORDS_POLLED = "RECORDS_POLLED",
  RECORDS_SYNCED = "RECORDS_SYNCED",
  FAILURE_RECORDS_PROCESSED = "FAILURE_RECORDS_PROCESSED",
}

export const PipelineRunStageLabel: any = {
  [PipelineRunStage.RUN_TRIGGERED]: "Run Triggered",
  [PipelineRunStage.RECORDS_POLLED]: "Records Polled",
  [PipelineRunStage.RECORDS_SYNCED]: "Records Synced",
  [PipelineRunStage.FAILURE_RECORDS_PROCESSED]: "Failure Records Processed",
};

File: container/src/main/java/io/castled/models/PipelineRunStage.java
package io.castled.models;

public enum PipelineRunStage {
    RUN_TRIGGERED,
    RECORDS_POLLED,
    RECORDS_SYNCED,
    FAILURE_RECORDS_PROCESSED;

    public boolean recordsPolled() {
        return this == RECORDS_POLLED || this == RECORDS_SYNCED || this == FAILURE_RECORDS_PROCESSED;
    }
}
